### PR TITLE
Add new mechanism to upgrade system during installation

### PR DESCRIPTION
**Pull request by yanan-fu**
id: 1724985

Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
Any update for kickstart file ?

File: virttest/tests/unattended_install.py [hunk 1]
```diff
@@ -394,6 +394,11 @@ def answer_kickstart(self, answer_path):
             raise ValueError("Unexpected installation medium %s" % self.url)
         contents = re.sub(dummy_medium_re, content, contents)
 
+        dummy_rh_system_stream_id_re = r'\bRH_SYSTEM_STREAM_ID\b'
+        if re.search(dummy_rh_system_stream_id_re, contents):
+            rh_system_stream_id = self.params.get("rh_system_stream_id", "")
```
Comment (xutian) [inline virttest/tests/unattended_install.py]: Please use params["rh_system_stream_id"] instead self.params.get("rh_system_stream_id", ""),  If it not replace, that may cause install fail.

**Comment by luckyh** [comment]
@xuma request for your review, thanks.

File: virttest/tests/unattended_install.py (hunk 1 above)
Comment (yanan-fu) [reply virttest/tests/unattended_install.py]: @xutian  some guest do not need to do system upgrade, then no param `rh_system_stream_id` is ok,  then we replace `RH_SYSTEM_STREAM_ID`  will  ''.
In kickstart, we do system upgrade according to the value of stream id.

**Comment by maxujun** [comment]
LGTM

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**